source/base/NexusApp.cc

```diff
@@ -77,13 +77,13 @@ NexusApp::NexusApp(G4String init_macro): G4RunManager(), gen_name_(""),
   // by the time we process the initialization macro.
 
   // The physics lists are handled with Geant4's own 'factory'
-  physicsList = new G4GenericPhysicsList();
+  auto pl = std::make_unique<G4GenericPhysicsList>();
 
   auto batch = std::make_unique<BatchSession>(init_macro.c_str());
   batch->SessionStart();
 
   // Set the physics list in the run manager
-  this->SetUserInitialization(physicsList);
+  this->SetUserInitialization(pl.release());
```

Comment:
The initialization of `pl` should not be separated from its use, by the completely unrelated `batch` code.

Lines 80 and 86 could, and probably should, be condensed down to nothing more than
```c++
this -> SetUserInitialization(new G4GenericPhysicsList());
```
which states very clearly that the lifetime of the physics list is `SetUserInitialization`'s problem, not ours.

In fact, this is a classic example of the only situation where `new` should appear in modern C++ code[*]: I am creating something and immediately giving ownership of it to a legacy consumer. 

[*]  unless you are implementing a low-level utility such as `std::unique_ptr` or `std::vector` itself, which you will not be doing in Nexus.

Reply:
In this case the order cannot be changed, since we need to instantiate a `G4GenericPhysicsList` first, then read the `init` configuration file, which contains the physics lists that needs to be registered and finally hand over to G4 our pointer through `SetUserInitialization`.

Reply:
So `batch->SessionStart()` results in some code being executed that, magically, behind the scenes, depends on the existence of a `G4GenericPhysicsList` instance whose existence has not been communicated anywhere (at least not in any way that is visible in this code)?

Genuis! (Well, I shouldn't be surprised: It's Geant4.)

From the perspective of someone reading the C++ code that is written here, these lines commute! If there is a reason why these lines do not commute, then it should be thoroughly and loudly documented right here, with unmissable comments in the code.

(Looks like another G4 trap that I should try to address in `nain4`.)